src/gui/shell.cpp

```diff
@@ -867,6 +866,38 @@ void Shell::handleGuiFontFunction(const QVariantList& args)
 	setGuiFont(fdesc, force, true /*setOption*/);
 }
 
+void Shell::handleCloseEvent(const QVariantList &args)
+{
+	qDebug() << "Neovim requested a GUI close";
+
+	// Decide which exit status Neovim-qt should exit with.
+	int status{};
+	if (args.size() == 1) {
+		// If there are no arguments, then the exit status is zero.
+		status = 0;
+	} else if (args.size() == 2) {
+		auto arg = args.at(1);
+
+		// Or, if there is an argument, then the exit status is
+		// specified in the argument.
+		bool ok{};
+		status = arg.toInt(&ok);
+		if (!ok) {
+			// If there is some conversion error, emit warning and ignore this
+			// event.  (Though it cannot prevent Neovim from closing.)
+			qWarning() << "Unexpected argument for a exit status: " << arg;
+			return;
+		}
+	} else {
```

Comment:
nitpick/optional: Generally having failures like this early on streamlines the code.

Ex) check for bad command format at the very beginning and exit on failure. This also prevents the need to check `ok`.